spec/dummy/spec/rails_helper.rb

```diff
@@ -76,21 +78,6 @@
 
   # Capybara config
   config.include Capybara::DSL
-  #
-  # selenium_firefox webdriver only works for Travis-CI builds.
-  default_driver = :selenium_chrome_headless
-
-  supported_drivers = %i[selenium_chrome_headless selenium_chrome selenium selenium_headless]
-  driver = ENV["DRIVER"].try(:to_sym).presence || default_driver
-  Capybara.javascript_driver = driver
-  Capybara.default_driver = driver
-
-  raise "Unsupported driver: #{driver} (supported = #{supported_drivers})" unless supported_drivers.include?(driver)
-
-  Capybara.register_server(Capybara.javascript_driver) do |app, port|
-    require "rack/handler/puma"
-    Rack::Handler::Puma.run(app, Port: port)
-  end
 
   config.before(:each, type: :system, js: true) do
     driven_by driver
```

Comment:
@G-Rath Tested this locally and it's failing 😅 Pretty sure it's that line here. After removals we are ending with nil so defaulting back to whatever driver is default 🤔 With the addition as suggestion below it does seem to work as expected though

```suggestion
    driven_by :selenium_chrome
```

Reply:
Ironically I'm pretty sure that's the most problematic:

```
2024-04-06 07:54:25 INFO Selenium [:logger_info] Details on how to use and modify Selenium logger:
  https://selenium.dev/documentation/webdriver/troubleshooting/logging

2024-04-06 07:54:25 WARN Selenium [:selenium_manager] The version of chrome cannot be detected. Trying with latest driver version
2024-04-06 07:54:26 WARN Selenium [:selenium_manager] The version of chrome cannot be detected. Trying with latest driver version
F2024-04-06 07:54:26 WARN Selenium [:selenium_manager] The version of chrome cannot be detected. Trying with latest driver version
2024-04-06 07:54:27 WARN Selenium [:selenium_manager] The version of chrome cannot be detected. Trying with latest driver version
F

Failures:

  1) hello_world loads correct view
     Got 0 failures and 2 other errors:

     1.1) Failure/Error: visit "/hello_world"

          Selenium::WebDriver::Error::UnknownError:
            unknown error: cannot find Chrome binary
```

I'll do some more digging - for context, WSL (Windows Subsystem for Linux) runs an integrated terminal-based VM within Windows which nowdays ships with WSLg by default so you can run GUIs natively alongside Windows apps; I've been able to run both headless and headful browser based tests with RSpec without major issue though I already know it doesn't work in all cases but at this point I think that's more a configuration thing then WSL itself.

i.e. with apps generated with our [`rails-template`](https://github.com/ackama/rails-template) I can run standard heavy tests, but our lighthouse accessibility and performance based tests which uses [`lighthouse-matchers`](https://github.com/ackama/lighthouse-matchers) fail

Reply:
Interesting. On my end without this change, the JS specs were falling as there was nothing executing JS. If they passed on your end, then I wonder what is running that 🤔

Reply:
Ok apparently I've just been silly - updating `selenium-webdriver` was the real fix, because they introduced the webdriver autofinding logic in ~4.15+ and I was using 4.10 🤷